backend/src/main/kotlin/org/loculus/backend/utils/MetadataEntry.kt

```diff
@@ -17,31 +18,37 @@ fun metadataEntryStreamAsSequence(metadataInputStream: InputStream): Sequence<Me
         CSVFormat.TDF.builder().setHeader().setSkipHeaderRecord(true).build(),
     )
 
-    if (!csvParser.headerNames.contains(HEADER_TO_CONNECT_METADATA_AND_SEQUENCES)) {
-        throw UnprocessableEntityException(
-            "The metadata file headers do not contain the header '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES': ${csvParser.headerNames}",
+    val headerNames = csvParser.headerNames
+    val submissionIdHeaders = listOf(
+        HEADER_TO_CONNECT_METADATA_AND_SEQUENCES,
+        HEADER_TO_CONNECT_METADATA_AND_SEQUENCES_ALTERNATE_FOR_BACKCOMPAT,
+    ).filter { headerNames.contains(it) }
+
+    when {
+        submissionIdHeaders.isEmpty() -> throw UnprocessableEntityException(
+            "The metadata file does not contain either header '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES' or '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES_ALTERNATE_FOR_BACKCOMPAT'",
+        )
+        submissionIdHeaders.size > 1 -> throw UnprocessableEntityException(
+            "The metadata file contains both '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES' and '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES_ALTERNATE_FOR_BACKCOMPAT'. Only one is allowed.",
         )
     }
+    val submissionIdHeader = submissionIdHeaders.first()
```

Comment:
Why use first here rather than explicit `HEADER_TO_CONNECT_METADATA_AND_SEQUENCES`?

I see, because this should only contain one item after filtering out the other one.